include/simdjson/generic/numberparsing.h

```diff
@@ -608,12 +608,15 @@ simdjson_unused simdjson_really_inline simdjson_result<uint64_t> parse_unsigned(
 
   // If there were no digits, or if the integer starts with 0 and has more than one digit, it's an error.
   int digit_count = int(p - start_digits);
-  if (digit_count == 0 || ('0' == *start_digits && digit_count > 1)) { return NUMBER_ERROR; }
-  if (!jsoncharutils::is_structural_or_whitespace(*p)) { return NUMBER_ERROR; }
+  if (digit_count == 0) { return INCORRECT_TYPE; }
+  if ('0' == *start_digits && digit_count > 1) { return NUMBER_ERROR; }
+  if (!jsoncharutils::is_structural_or_whitespace(*p)) {
```

Comment:
Do we want simdjson_unlikely around this check?  What is inside the 'if' clause is not going to be cheap with this change. (May not help.)